spec/reports/subscription_collection_report_spec.rb

```diff
@@ -0,0 +1,204 @@
+require 'rails_helper'
+require 'csv'
+
+RSpec.describe SubscriptionCollectionReport do
```

Comment:
You could simplify the specs for this class doing some comparsion of pre-generated reports with some generated by the suite